### PR TITLE
IBP-5520 post lists implementation

### PR DESCRIPTION
**Pull request by mderamos**

### CODE REVIEW

File: src/main/java/org/generationcp/middleware/dao/germplasmlist/GermplasmListDAO.java [hunk 1]
```diff
@@ -948,8 +940,8 @@ public void appendListsFromQuery(final StringBuilder stringBuilder) {
 	}
 
 	public void appendListSeachFilters(final StringBuilder stringBuilder, final GermplasmListSearchRequestDTO requestDTO) {
```
Comment (darla-leafnode) [inline src/main/java/org/generationcp/middleware/dao/germplasmlist/GermplasmListDAO.java:942]: correct typo (existing in master): appendListSea**r**chFilters

File: src/main/java/org/generationcp/middleware/domain/search_request/brapi/v2/GermplasmListSearchRequestDTO.java [hunk 2]
```diff
@@ -4,21 +4,23 @@
...
 	private String listType;
 	private String listName;
-	private String listDbId;
+	private List<String> listDbIds;
 	private String listSource;
 	private String externalReferenceID;
 	private String externalReferenceSource;
 
-	public GermplasmListSearchRequestDTO(final String listType, final String listName, final String listDbId, final String listSource,
+	public GermplasmListSearchRequestDTO(final String listType, final String listName, final List<String> listDbId, final String listSource,
```
Comment (darla-leafnode) [inline src/main/java/org/generationcp/middleware/domain/search_request/brapi/v2/GermplasmListSearchRequestDTO.java:19]: Rename param as `listDbIds`

File: src/test/java/org/generationcp/middleware/brapi/GermplasmListServiceBrapiImplTest.java [hunk 3]
```diff
@@ -64,22 +74,58 @@ public void testSearchGermplasmListDTOs() {
...
+		Assert.assertEquals(1, germplasmListDTOList.size());
+		Assert.assertEquals(importRequestDTO.getListName(), germplasmListDTOList.get(0).getListName());
+		Assert.assertEquals(importRequestDTO.getListDescription(), germplasmListDTOList.get(0).getListDescription());
+		Assert.assertEquals(importRequestDTO.getDateCreated(),
+			Util.formatDateAsStringValue(germplasmListDTOList.get(0).getDateCreated(), Util.FRONTEND_DATE_FORMAT));
+		Assert.assertEquals(importRequestDTO.getListOwnerPersonDbId(), germplasmListDTOList.get(0).getListOwnerPersonDbId());
+		Assert.assertEquals(1, germplasmListDTOList.get(0).getExternalReferences().size());
+		Assert.assertEquals(1, germplasmListDTOList.get(0).getListSize().intValue());
+		final ExternalReferenceDTO savedExternalReferenceDTO = germplasmListDTOList.get(0).getExternalReferences().get(0);
+		Assert.assertEquals(externalReferenceDTO.getReferenceID(), savedExternalReferenceDTO.getReferenceID());
```
Comment (darla-leafnode) [inline src/test/java/org/generationcp/middleware/brapi/GermplasmListServiceBrapiImplTest.java:128]: Add some assertions related to the germplasm list data saved

**Review by darla-leafnode — CHANGES_REQUESTED** [review]

File: src/main/java/org/generationcp/middleware/api/brapi/GermplasmListServiceBrapiImpl.java [hunk 4]
```diff
@@ -57,4 +90,84 @@ public long countGermplasmListDTOs(final GermplasmListSearchRequestDTO searchReq
...
+			this.saveGermplasmListData(germplasmList, importRequestDTO.getData(), crossExpansions, plotCodeValuesByGIDs, germplasmDTOMap,
+				preferredNamesMap, namesByGid);
+			savedListIds.add(germplasmList.getId().toString());
+		}
+		final GermplasmListSearchRequestDTO germplasmListSearchRequestDTO = new GermplasmListSearchRequestDTO();
+		germplasmListSearchRequestDTO.setListDbIds(savedListIds);
+		return this.searchGermplasmListDTOs(germplasmListSearchRequestDTO, null);
+	}
+
+	private void saveGermplasmListData(final GermplasmList germplasmList, final List<String> guids,
```
[the remark this answers is not in the document]
Comment (mderamos) [reply src/main/java/org/generationcp/middleware/api/brapi/GermplasmListServiceBrapiImpl.java:127]: GUIDs are the values contained in the data field. The guid value is used to get the germplasm from the map.

**Review by darla-leafnode — APPROVED** [review]

**Pull request merged**